src/trajectory.jl

```diff
@@ -246,13 +254,29 @@ function randcat(rng, zs::AbstractVector{<:PhasePoint}, unnorm_ℓP::AbstractMat
 end
 
 function samplecand(rng, τ::StaticTrajectory{MultinomialTS}, h, z)
-    zs = step(τ.integrator, h, z, τ.n_steps; res=[z for _ in 1:abs(τ.n_steps)])
+    n_steps = abs(τ.n_steps)
+    n_fwd = rand(0:n_steps) # FIXME: deal with multi-chain generically
+    zs_fwd = step(τ.integrator, h, z, n_fwd; fwd=true, res=[z for _ in 1:n_fwd])
+    n_bwd = n_steps - n_fwd
+    zs_bwd = step(τ.integrator, h, z, n_bwd; fwd=false, res=[z for _ in 1:n_bwd])
+    zs = vcat(reverse(zs_bwd)..., z, zs_fwd...)
     ℓws = -energy.(zs)
     if eltype(ℓws) <: AbstractVector
-        ℓws = hcat(ℓws...)
+        ℓws = cat(ℓws...; dims=2)
     end
     unnorm_ℓprob = ℓws
-    return randcat(rng, zs, unnorm_ℓprob)
+    z′ = randcat(rng, zs, unnorm_ℓprob)
+    # Computing adaptation statistics for dual averaging as done in NUTS
+    Hs = -ℓws
+    ΔH = Hs .- energy(z)
+    α = exp.(min.(0, -ΔH))  # this is a matrix for vectorized mode and a vector otherwise
+    α =
```

Comment:
Consider `α = typeof(α) <: AbstractVector ? mean(α) : vec(mean(α; dims=2))` for clarity.

Reply:
Or at least putting the if-statement on the same line as `=`, i.e. `α = if typeof(α) <: AbstractVector ...` (subjective opinion)

Reply:
Sure. I will take Hong's suggestion. What I currently have here is a very bad style!